src/Orc.CsvHelper/Orc.CsvHelper.Shared/Services/CsvReaderService.cs

```diff
@@ -117,15 +110,47 @@ public CsvReader CreateReader(string csvFilePath, CsvClassMap csvMap, CsvConfigu
 
             return csvReader;
         }
+        #endregion
 
-        private CsvReader CreateReaderCore(string csvFilePath, CsvConfiguration csvConfiguration = null, CultureInfo culture = null)
+        #region Methods
+        protected virtual IEnumerable<T> ReadData<T>(string csvFilePath, Action<T> initializer, bool throwOnError, CsvReader csvReader)
         {
-            if (csvConfiguration != null && culture != null)
+            var items = new List<T>();
+
+            try
             {
-                csvConfiguration.CultureInfo = culture;
+                while (csvReader.Read())
+                {
+                    var record = csvReader.GetRecord<T>();
+                    initializer?.Invoke(record);
```

Comment:
I think using .Invoke is more expensive than checking for null. In the future we will create benchmarks for this.